### ISSUE TITLE
ISubscriber no longer mockable

### ISSUE THREAD

**Issue by BrennanConroy** [StackExchange/StackExchange.Redis#969]
With the introduction of `ChannelMessageQueue Subscribe(...)` and the async version, `ISubscriber` is no longer mockable (because of `sealed class ChannelMessageQueue`) so it's difficult/impossible to unit test code that uses Redis.

**Comment by mgravell** [comment]
Fair point. I'll think about how we can re-expose this.

**Comment by anurse** [comment]
A really awesome way to expose this for testing would be if we could wrap a `ChannelReader<RedisMessage>` up into a `ChannelMessageQueue`

**Comment by anurse** [comment]
Any update here? We'd like to be able to restore our testing here :). We're happy to help contribute something if it makes sense. My initial design thought is to make it possible by having `ChannelMessageQueue` be constructable from a `ChannelReader<ChannelMessage>`

**Comment by hansarnevartdal** [comment]
Would at least be beneficial to enable unit testing on the channel message handler called by the channel message queue, but this uses the ChannelMessage struct with internal ctor, so not really doable as is.

**Comment by viktors-telle** [comment]
Are there any plans to resolve this issue?

**Comment by kmcclellan** [comment]
Agreed. Is the problem with extracting an `IChannelMessageQueue` simply because it is breaking? I'd love to see that in v3.

**Comment by kmcclellan** [comment]
FWIW this isn't just about testing. I had wanted to implement an ISubscriber that wraps an inner subscriber using Polly retries, and was stymied by this limitation.
